BraveRewardsUI/Wallet/WalletViewController.swift

```diff
@@ -111,27 +111,47 @@ class WalletViewController: UIViewController, RewardsSummaryProtocol {
     
     walletView.headerView.layoutIfNeeded()
     walletView.contentView?.layoutIfNeeded()
+    walletView.rewardsSummaryView.layoutIfNeeded()
     
     guard let contentView = walletView.contentView else { return }
-    
-    var height: CGFloat = walletView.headerView.bounds.height + walletView.rewardsSummaryView.rewardsSummaryButton.bounds.height
-    if let scrollView = walletView.contentView?.innerScrollView {
-      scrollView.contentInset = UIEdgeInsets(top: walletView.headerView.bounds.height, left: 0, bottom: 0, right: 0)
-      scrollView.scrollIndicatorInsets = scrollView.contentInset
-      
-      height += scrollView.contentSize.height
-    } else {
-      height += contentView.bounds.height
-    }
-    
+    var height: CGFloat
     if state.ledger.isEnabled {
-      height = RewardsUX.preferredPanelSize.height
+      if summaryRows.isEmpty && isLocal {
+        height = walletView.rewardsSummaryView.scrollView.contentSize.height +
+          walletView.headerView.bounds.height +
+          walletView.rewardsSummaryView.rewardsSummaryButton.bounds.height +
+          walletView.rewardsSummaryView.monthYearLabel.bounds.height +
+          40 // Padding in Wallet view hierarchy 
+      } else {
+        if let scrollView = walletView.contentView?.innerScrollView {
+          scrollView.contentInset = UIEdgeInsets(top: walletView.headerView.bounds.height, left: 0, bottom: 0, right: 0)
+          scrollView.scrollIndicatorInsets = scrollView.contentInset
+          // Setting content Offset so that scrollview is properly aligned in smaller devices
+          scrollView.contentOffset.y = -scrollView.contentInset.top
+        }
+        height = RewardsUX.preferredPanelSize.height
+      }
+    } else {
+      height = walletView.headerView.bounds.height + walletView.rewardsSummaryView.rewardsSummaryButton.bounds.height
+      if let scrollView = walletView.contentView?.innerScrollView {
+        scrollView.contentInset = UIEdgeInsets(top: walletView.headerView.bounds.height, left: 0, bottom: 0, right: 0)
+        scrollView.scrollIndicatorInsets = scrollView.contentInset
+        
+        height += scrollView.contentSize.height
+      } else {
+        height += contentView.bounds.height
+      }
     }
     
     let newSize = CGSize(width: RewardsUX.preferredPanelSize.width, height: height)
     if preferredContentSize != newSize {
       preferredContentSize = newSize
     }
+    if isLocal {
+      transformRewardSummaryView(isExpanding: true, animated: false)
+      walletView.rewardsSummaryView.rewardsSummaryButton.slideToggleImageView.image = nil
```

Comment:
You should override `isDisabled` on `RewardsSummaryButton` and _hide_ the image (instead of setting it to nil) based on the state